src/playermat/Playmat.ttslua

```diff
@@ -424,32 +409,54 @@ end
 
 function markSpawned(object)
   local saved = DATA_HELPER.call('setSpawnedPlayerCardGuid', {object.getGUID(), true})
-  if not saved then
-    error('attempt to mark player card spawned before data loaded')
-  end
+  if not saved then error('attempt to mark player card spawned before data loaded') end
 end
 
+-- contains position and amount of boxes for the upgradesheets that change uses
+-- Alchemical Distillation, Damning Testimony, Living Ink and Hyperphysical Shotcaster
+local customizationsTable = {
+  ["09040"] = {5, 2},
+  ["09059"] = {2, 2},
+  ["09079"] = {3, 2},
+  ["09119"] = {6, 4}
+}
+
 function spawnTokensFor(object)
   local cardMetadata = JSON.decode(object.getGMNotes()) or {}
-  local token = nil
   local type = nil
+  local token = nil
   local tokenCount = 0
-  if (cardMetadata.uses ~= nil) then
-    for i, useInfo in ipairs(cardMetadata.uses) do
-      token = useInfo.token
+  if cardMetadata.uses ~= nil then
+    for _, useInfo in ipairs(cardMetadata.uses) do
       type = useInfo.type
+      token = useInfo.token
       tokenCount = useInfo.count
-      if (activeInvestigatorId == "03004" and useInfo.type == "Charge") then
-        tokenCount = tokenCount + 1
+    end
+
+    -- additional uses for certain customizable cards (by checking the upgradesheets)
+    if customizationsTable[cardMetadata.id] ~= nil then
+      for _, obj in ipairs(searchArea(PLAY_ZONE_POSITION, PLAY_ZONE_SCALE)) do
+        local obj = obj.hit_object
+        if obj.tag == "Card" then
+          local notes = JSON.decode(obj.getGMNotes()) or {}
+          if notes.id == (cardMetadata.id .. "-c") then
+            local pos = customizationsTable[cardMetadata.id][1]
+            local boxes = customizationsTable[cardMetadata.id][2]
+            if obj.getVar("markedBoxes")[pos] == boxes then tokenCount = tokenCount + 2 end
```

Comment:
We should try to avoid getVar() whenever possible.  It exposes the internal structure of an object, which makes changing that very problematic.  Instead, create a method in the target object and use .call().  That creates a layer of indirection where you can update the internals.

Wouldn't do it now, PR is already massive as it is, but consider for a near-future enhancement.

Reply:
Good idea, would need to update the upgradesheets for that (and also use require for them)